### PR TITLE
tock-rt0 crate

### PR DESCRIPTION
**Pull request by niklasad1**
### Pull Request Overview

This pull request removes the old spaghetti-rust code for nrf51 & nrf52 `init_data_segment` and `zero_bss`. As this was duplicated for all chips I moved it out to a separate crate and removed it for all chip crates should be tested to be on safe side!

This `crt0s` assumes 32-bit CPUs but could easily be generalized if needed.

### Testing Strategy

Tested on:
- [x] nrf52dk
- [x] nrf51dk
- [x] hail
- [ ] imix
- [ ] ek-tm4c1294xl
- [ ] launchxl

### TODO or Help Wanted

This pull request still needs...

### Documentation Updated

- [ ] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make formatall`.

### CODE REVIEW

**Comment by bradjc** [comment]
This is cool (and I'm always a fan when removed lines > added lines). I do think it needs a readme though.

Also, and others should chime in, but calling this _c_ rt seems weird to me.

**Comment by ppannuto** [comment]
Re naming, fair point, looks like japaric called his just "r0":
https://github.com/japaric/r0 :)

**Comment by rajivr** [comment]
> Feels like I'm re-inventing the wheel a bit here when his crate does everything I want but I guess we still have a policy against external crates?

I am new to Tock, but policy against external crates for the kernel seems like a good idea to me. It would make `git bisect` straight forward.

I've also also started using `tock-registers` outside of Tock, so the current policy looks to be good.

**Comment by ppannuto** [comment]
Tock-r0 sounds good to me

**Comment by ppannuto** [comment]
Following the logic the `crt0` is the C initial runtime, I went with `tock-rt0`. Also rebased on latest master to resolve the lockfile conflicts.

This should be good to go now.

**Comment by niklasad1** [comment]
Sorry, got caught up in other things! 

Thanks  @ppannuto 

Feels weird to approve/review my own PR but if somebody can have review/approve it and that would be good!

File: libraries/tock-rt0/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+//! Tock generic initial runtime (`rt0`) helpers
+
+#![no_std]
+
+/// Initializes the `static data`, by copying it into memory (RAM) from
+/// non-volatile memory (Flash)
+// Relocate data segment.
+// Assumes data starts right after text segment as specified by the linker
+pub unsafe fn init_data(mut edata: *mut u32, mut sdata: *mut u32, sdata_end: *mut u32) {
```
Comment (bradjc) [inline libraries/tock-rt0/src/lib.rs]: Seems odd to have `edata` as the argument here. Are these really `start_data_flash`, `start_data_ram`, `end_data_ram`?

Comment (bradjc) [reply libraries/tock-rt0/src/lib.rs]: Right, so let's just rename them so that future developers don't have to back calculate that mapping.

**Pull request merged**